packages/augur-core/src/contracts/trading/erc20proxy1155/ERC20Proxy1155.sol

```diff
@@ -0,0 +1,53 @@
+import 'ROOT/libraries/CloneFactory.sol';
+import 'ROOT/libraries/token/IERC20.sol';
+import 'ROOT/libraries/token/IERC1155.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+
+import 'ROOT/trading/erc20proxy1155/IERC20Proxy1155Nexus.sol';
+import 'ROOT/trading/erc20proxy1155/IERC20Proxy1155.sol';
+
+
+contract ERC20Proxy1155 is IERC20Proxy1155 {
+    uint8 constant public decimals = 18;
+    IERC20Proxy1155Nexus nexus;
+    uint256 public tokenId;
+
+    function initialize(IERC20Proxy1155Nexus _nexus, uint256 _tokenId) public {
+        nexus = _nexus;
+        tokenId = _tokenId;
+    }
+
+    function totalSupply() external view returns (uint256) {
+        return nexus.target1155().totalSupply(tokenId);
+    }
+
+    function balanceOf(address _account) public view returns (uint256) {
+        return nexus.target1155().balanceOf(_account, tokenId);
+    }
+
+    function transfer(address _to, uint256 _amount) public returns (bool) {
+        return nexus.transfer(this, msg.sender, _to, _amount);
+    }
+
+    function transferFrom(address _from, address _to, uint256 _amount) public returns (bool) {
+        return nexus.transferFrom(this, msg.sender, _from, _to, _amount);
+    }
+
+    function approve(address _spender, uint256 _amount) public returns (bool) {
+        return nexus.approve(this, msg.sender, _spender, _amount);
+    }
+
+    function allowance(address _owner, address _spender) public view returns (uint256) {
+        return nexus.allowance(this, _owner, _spender);
+    }
+
+    function logApproval(address _owner, address _spender, uint256 _amount) public {
+        require(msg.sender == address(nexus));
+        emit Approval(_owner, _spender, _amount);
```

Comment:
These logs should also come from the nexus along with the tokenId. This way only one contract needs to be listened to in order to get these logs.

Reply:
Actually nevermind. This is just to sate the ERC20 standard. We dont even need to listen to these and technically noone does

Reply:
You can already get these logs from the ERC1155 contract though.